### DISCUSSION TITLE
How Can i Add a Specific File in My OS

### DISCUSSION THREAD

**Discussion by ahmedbarakat2007** [CosmosOS/Cosmos#2785] [Q&A] [answered]
i want to add a file to my os but the issue is i don't know how can i include it to the kernal or the iso so i can use it

**Comment by 9xbt** [comment, marked as answer]
Ok, so in the VS solution explorer right click the file you wanna embed and click properties. Then select `Embedded Resource` in the drop down box. Then add this to your using directive
```cs
using IL2CPU.API.Attribs;
```
Then add this inside your kernel class (or wherever you want to have the variable for using it)
```cs
[ManifestResourceStream(ResourceName = “Namespace.Folder.Filename.extension”)]
static readonly byte[] rawFileData;
```
If for example you add a text file you would do this to get it in a string
```cs
static readonly string fileData = new string(rawFileData);
```
And thats about it. Note that this code is pseudocode and its not tested but it should work if you change the necessary stuff

**Comment by Samma2009** [comment]
[there is a cosmos documantation page about adding resources ceck it out (click me)](https://cosmosos.github.io/articles/Kernel/ManifestResouceStream.html)

**Reply by 9xbt** [reply]
That too yeah i forgot about that article
